src/main/java/com/example/bookstore/dto/RegisterUserDto.java

```diff
@@ -0,0 +1,20 @@
+package com.example.bookstore.dto;
+
+import com.example.bookstore.validation.FieldMatch;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+
+@Data
+@FieldMatch(fields = {"password", "repeatPassword"})
+public class RegisterUserDto {
+    @NotBlank
+    private String email;
+    @NotBlank
+    private String password;
+    private String repeatPassword;
+    @NotBlank
+    private String firstName;
+    @NotBlank
+    private String lastName;
+    private String shippingAddress;

```

Comment:
Too poor validation

Reply:
Can you explain what's wrong exactly? 
NotBlank annotation checks for null value and its trimmed length, so I can't figure out what else is required here.